File: pathfinder-editmode/src/main/java/de/cubbossa/pathfinder/editmode/DefaultGraphEditorFactory.java
package de.cubbossa.pathfinder.editmode;

import de.cubbossa.pathfinder.PathFinder;
import de.cubbossa.pathfinder.PathFinderProvider;
import de.cubbossa.pathfinder.editor.GraphEditor;
import de.cubbossa.pathfinder.editor.GraphEditorFactory;
import de.cubbossa.pathfinder.group.NodeGroup;
import de.cubbossa.pathfinder.PathFinderPlugin;
import de.cubbossa.pathfinder.editmode.renderer.EdgeArmorStandRenderer;
import de.cubbossa.pathfinder.editmode.renderer.EdgeEntityRenderer;
import de.cubbossa.pathfinder.editmode.renderer.NodeArmorStandRenderer;
import de.cubbossa.pathfinder.editmode.renderer.NodeEntityRenderer;
import de.cubbossa.pathfinder.editmode.renderer.NodeGroupListRenderer;
import de.cubbossa.pathfinder.editmode.renderer.ParticleEdgeRenderer;
import de.cubbossa.pathfinder.util.Version;
import org.bukkit.entity.Player;
import org.pf4j.Extension;

@Extension(points = GraphEditorFactory.class)
public class DefaultGraphEditorFactory implements GraphEditorFactory {

  @Override
  public GraphEditor<Player> createGraphEditor(NodeGroup group) {
    PathFinder pathFinder = PathFinderProvider.get();
    DefaultGraphEditor editor = new DefaultGraphEditor(group);

    editor.getRenderers().add(new ParticleEdgeRenderer(pathFinder.getConfiguration().getEditMode()));
    try {
      if (new Version(PathFinderPlugin.getInstance().getServer().getBukkitVersion().split("-")[0]).compareTo(new Version("1.19.4")) >= 0) {
        editor.addRenderer(new NodeGroupListRenderer(PathFinderPlugin.getInstance(), 15, 8));
        editor.addRenderer(new NodeEntityRenderer(PathFinderPlugin.getInstance()));
        editor.addRenderer(new EdgeEntityRenderer(PathFinderPlugin.getInstance()));
      } else {
        editor.addRenderer(new NodeArmorStandRenderer(PathFinderPlugin.getInstance()));
        editor.addRenderer(new EdgeArmorStandRenderer(PathFinderPlugin.getInstance()));
      }
    } catch (Throwable t) {
      t.printStackTrace();
    }

    return editor;
  }
}

File: pathfinder-editmode/src/main/kotlin/de/cubbossa/pathfinder/editmode/DefaultGraphEditorFactory.kt
package de.cubbossa.pathfinder.editmode

import de.cubbossa.pathfinder.PathFinder
import de.cubbossa.pathfinder.PathFinderPlugin
import de.cubbossa.pathfinder.editmode.renderer.*
import de.cubbossa.pathfinder.editor.GraphEditor
import de.cubbossa.pathfinder.editor.GraphEditorFactory
import de.cubbossa.pathfinder.group.NodeGroup
import de.cubbossa.pathfinder.util.Version
import org.bukkit.entity.Player
import org.pf4j.Extension

@Extension(points = [GraphEditorFactory::class])
class DefaultGraphEditorFactory : GraphEditorFactory {
    override fun createGraphEditor(group: NodeGroup): GraphEditor<Player> {
        val pathFinder = PathFinder.get()
        val editor = DefaultGraphEditor(group)
        val plugin = PathFinderPlugin.getInstance()

        editor.renderers.add(ParticleEdgeRenderer(pathFinder.configuration.editMode))
        try {
            if (Version(
                    PathFinderPlugin.getInstance().server.bukkitVersion.split("-".toRegex())
                        .dropLastWhile { it.isEmpty() }.toTypedArray()[0]
                ).compareTo(Version("1.19.4")) >= 0
            ) {
                editor.addRenderer(NodeGroupListRenderer(plugin, 15.0, 8.0))
                editor.addRenderer(NodeEntityRenderer(plugin))
                editor.addRenderer(EdgeEntityRenderer(plugin))
            } else {
                editor.addRenderer(NodeArmorStandRenderer(plugin))
                editor.addRenderer(EdgeArmorStandRenderer(plugin))
            }
        } catch (t: Throwable) {
            t.printStackTrace()
        }
        return editor
    }
}
